src/buffer.h

```diff
@@ -157,6 +157,15 @@ class Buffer {
   /// |element_count| elements
   void SetSizeInBytes(uint32_t size_in_bytes);
 
+  /// max_size_in_bytes_ is the total size in bytes needed to hold the buffer
```

Comment:
This is a _definition_ of the max_size_in_bytes_ field, which should go just before the member declaration at line 200.  

Separately, there's a fragile disconnect between setting max_size_in_bytes_, element_count_ and resizing the bytes_ vector.   No single one of them is the source of truth.
You'll need to write comments at their member declarations to define exactly what they mean, and therefore determine the single source of truth.  To be clear, this is a continuation of a previously existing problem with this class definition.

Reply:
I can predict this can get complicated so I filed issue #599